utils/obsproc/Viirsaod2Ioda.h

```diff
@@ -132,12 +121,80 @@ namespace gdasapp {
                 }
                 obserror[i][j] = obserrorValue;
                 mask[i][j] = 1;
-                nobs += 1;
              }
           }
         }
       }
 
+      // superobing, not work yet...
+      std::vector<std::vector<float>> obsvalue_s;
+      std::vector<std::vector<float>> lon2d_s;
+      std::vector<std::vector<float>> lat2d_s;
+      std::vector<std::vector<float>> obserror_s;
+      std::vector<std::vector<int>> mask_s;
+
+      if ( fullConfig_.has("binning") ) {
+        // deal with longitude when points cross the international date line
+        float minLon;
+        // = std::numeric_limits<float>::max();
+        float maxLon;
+        // = std::numeric_limits<float>::min();
+
+        for (const auto& row : lon) {
+           minLon = std::min(minLon, *std::min_element(row.begin(), row.end()));
+           maxLon = std::max(maxLon, *std::max_element(row.begin(), row.end()));
+        }
+
+        if (maxLon - minLon > 180) {
+        // Normalize longitudes to the range [0, 360)
+           for (auto& row : lon) {
+               for (float& lonValue : row) {
+                   lonValue = fmod(lonValue + 360, 360);
+               }
+           }
+        }
+
+        lon2d_s = gdasapp::superobutils::subsample2D(lon, mask, fullConfig_);
+        for (auto& row : lon2d_s) {
+            for (float& lonValue : row) {
+                lonValue = fmod(lonValue + 360, 360);
+            }
+        }
+
+
+        lat2d_s = gdasapp::superobutils::subsample2D(lat, mask, fullConfig_);
+        mask_s = gdasapp::superobutils::subsample2D(mask, mask, fullConfig_);
+        // implement weighted average
+        if (fullConfig_.has("binning.cressman radius")) {
+        // obsvalue_s = gdasapp::superobutils::subsample2D_cressman(obsvalue, lat, lon,
```

Comment:
maybe a comment here about how this isn't working yet?

Reply:
comment added.